mlflow/pytorch/torchserve/deploy/serve.py

```diff
@@ -107,19 +117,47 @@ def update_deployment(self, name, model_uri=None, flavor=None, config=None):
         resp = requests.put(url)
 
         if resp.status_code != 202:
-            raise Exception("Unable to list deployments")
+            raise Exception(
+                "Unable to update deployment with name %s. Server returned status code %s and response: %s"
+                % (name, resp.status_code, resp.content)
+            )
         return {"flavor": flavor}
 
     def list_deployments(self):
         """
         List the names of all model deployments in the specified target. These names can be used with
         delete , update and get commands
         """
-        url = "{}/{}".format(self.management_api, "models")
-        resp = requests.get(url)
-        if resp.status_code != 200:
-            raise Exception("Unable to list deployments")
-        return [resp.text]
+
+        deployment_list = []
+        limit = self.default_limit
+        nextPageToken = 0
+
+        while True:
+            url = "{}/{}?{}={}&{}={}".format(
```

Comment:
Could we use the `params` arg to `requests.get` to pass query parameters like `next_page_token` and `limit`, instead of injecting them into the URL itself? See https://requests.readthedocs.io/en/master/user/quickstart/#passing-parameters-in-urls